Sources/KeyboardShortcuts/KeyboardShortcuts.swift

```diff
@@ -437,11 +471,54 @@ public enum KeyboardShortcuts {
 	public static func onKeyUp(for name: Name, action: @escaping () -> Void) {
 		legacyKeyUpHandlers[name, default: []].append(action)
 		registerShortcutIfNeeded(for: name)
+
+		// observe changes to the UserDefaults instance for the given shortcut name
+		startObservingShortcutIfNeeded(for: name)
 	}
 
 	private static let userDefaultsPrefix = "KeyboardShortcuts_"
 
-	private static func userDefaultsKey(for shortcutName: Name) -> String { "\(userDefaultsPrefix)\(shortcutName.rawValue)"
+	private static func userDefaultsKey(for shortcutName: Name) -> String {
+		"\(userDefaultsPrefix)\(shortcutName.rawValue)"
+	}
+
+	/**
+	Start observing changes to the `UserDefaults` instance for a specific shortcut name.
+
+	This function manages the lifecycle of observations for keyboard shortcuts in the given `UserDefaults` instance (set by `userDefaults` property):
+	- Checks if the shortcut is already being observed
+	- If already observed, restarts the observation
+	- If not observed, creates a new observation and adds it to the observers list
+	
+	The observation handles changes to the shortcut configuration in the suite:
+	- When the shortcut is removed (value becomes nil), unregisters the shortcut
+	- When the shortcut is added or modified, registers the new shortcut
+	
+	- Parameter name: The name of the shortcut to observe
+	*/
+	private static func startObservingShortcutIfNeeded(for name: Name) {
+		let key = userDefaultsKey(for: name)
+
+		// check userDefaultsObservers to see if we are already observing this key
+		if let observer = userDefaultsObservers.first(where: { $0.key == key }) {
+			observer.start()
```

Comment:
I think this could use a better code comment because just looking at it, it's not immediate obvious why we may want to call start() on an already existing observation.